### PR TITLE
Host macro idempotency

### LINKED ISSUE #715: zabbix_host: Handling of macros is not idempotent
##### SUMMARY
I run task for update host macros. When I run this task again, it will changed again.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
community.zabbix.zabbix_host
##### ANSIBLE VERSION

```
ansible [core 2.12.6]
  config file = /home/vagrant/ansible/ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/dist-packages/ansible
  ansible collection location = /home/vagrant/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True

```

##### CONFIGURATION
```
HOST_KEY_CHECKING(/home/vagrant/ansible/ansible.cfg) = False

```

##### OS / ENVIRONMENT / Zabbix Version

Zabbix-server: Debian 11, zabbix version 6.0, DB postgres
Zabbix-agent: Debian 10, zabbix-agent2 version 6.0

##### STEPS TO REPRODUCE

```yaml
- name: "New macros apply"
  community.zabbix.zabbix_host:
    server_url: "{{ zabbix_server_url_protocol}}{{ zabbix_server_url_hostname }}"
    login_user: "{{ zabbix_server_login_user }}"
    login_password: "{{ zabbix_server_login_password }}"
    host_name: "{{ zabbix_agent_config_Hostname }}"
    visible_name: "{{ zabbix_agent_config_Hostname }}"
    timeout: 25
    force: false
    macros:
      -  macro: '{$SMARTCTL_STATIC_DISKS}'
         value: "test"
    status: enabled
    state: present
```

##### EXPECTED RESULTS

It will changed only once, at first run.

##### ACTUAL RESULTS

It is always cnanged.

### PR DESCRIPTION
**Pull request by BGmot**
##### SUMMARY
Fixes #715 

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by BGmot** [comment]
Is there any concern about this PR?

**Comment by D3DeFi** [comment]
> Is there any concern about this PR?

Sorry, havent had that much time lately. Will review this in a couple of hours and merge even when we didnt get feedback from the related issue

File: tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml [hunk 1]
```diff
@@ -1357,10 +1374,10 @@
             type: secret
       register: zabbix_host1
 
-    - name: expect to succeed and that things have not changed
+    - name: expect to succeed and that things have changed as it has secret value
       assert:
         that:
-          - "zabbix_host1 is not changed"
+          - "zabbix_host1 is changed"
```
Comment (D3DeFi) [inline tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml:1380]: I think we may face streak of users that will complain about behavioral change. I am not particularly against thins change though

File: plugins/modules/zabbix_host.py [hunk 2]
```diff
@@ -1085,9 +1084,9 @@ def main():
                     macro.pop('type')
                 else:
                     if macro['type'] == 'text':
-                        macro['type'] = 0
+                        macro['type'] = '0'
```
Comment (D3DeFi) [inline plugins/modules/zabbix_host.py:1087]: Would this break some evaluations? For example one on line 812?

**Review by D3DeFi — CHANGES_REQUESTED** [review]
Thanks for the contribution. Once again, good job :) 

Please see my comments, we can merge then

File: tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml (hunk 1 above)
Comment (BGmot) [reply tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml:1380]: Logic is simple: it's not correct to provide 'not changed' if you have no idea whether you are changing the value or not. Moreover if we report 'not changed' then we don't even call host.update() thus your changes to secret macro will never be applied.

File: plugins/modules/zabbix_host.py (hunk 2 above)
Comment (BGmot) [reply plugins/modules/zabbix_host.py:1087]: We need to remove the whole block 809-813 as we don't want to `macro.pop('value', False)` if we follow the logic "secret macro value will have to be updated always" (see my comment above please)

File: tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml (hunk 1 above)
Comment (D3DeFi) [reply tests/integration/targets/test_zabbix_host/tasks/zabbix_host_tests.yml:1380]: Yes, I agree. Just pointing out that there will be definitely someone to open an issue about :)

File: plugins/modules/zabbix_host.py (hunk 2 above)
Comment (D3DeFi) [reply plugins/modules/zabbix_host.py:1087]: Ok, I guess it never being called is also a way to achieve that :) Feel free to cleanup in a follow up PR if you feel like it

Comment (BGmot) [reply plugins/modules/zabbix_host.py:1087]: Submitted https://github.com/ansible-collections/community.zabbix/pull/724

**Pull request merged**